IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationTokenRequest.m

```diff
@@ -63,6 +75,35 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)er
         if (!_configuration) return nil;
         
         _providerType = MSIDProviderTypeFromString([json msidStringObjectForKey:MSID_PROVIDER_TYPE_JSON_KEY]);
+        
+        _oidcScope = [json msidStringObjectForKey:MSID_BROKER_EXTRA_OIDC_SCOPES_KEY];
+        
+        NSString *extraQueryParam = [json msidStringObjectForKey:MSID_BROKER_EXTRA_QUERY_PARAM_KEY];
+        _extraQueryParameters = [NSDictionary msidDictionaryFromWWWFormURLEncodedString:extraQueryParam];
+        
+        _instanceAware = [json msidBoolObjectForKey:MSID_BROKER_INSTANCE_AWARE_KEY];
+        
+        NSString *enrollmentIdsStr = [json msidStringObjectForKey:MSID_BROKER_INTUNE_ENROLLMENT_IDS_KEY];
+        if (enrollmentIdsStr)
+        {
+            _enrollmentIds = (NSDictionary *)[[MSIDJsonSerializer new] fromJsonString:enrollmentIdsStr ofType:NSDictionary.class context:nil error:nil];
+        }
+        
+        NSString *mamResourcesStr = [json msidStringObjectForKey:MSID_BROKER_INTUNE_MAM_RESOURCE_KEY];
+        if (mamResourcesStr)
+        {
+            _mamResources = (NSDictionary *)[[MSIDJsonSerializer new] fromJsonString:mamResourcesStr ofType:NSDictionary.class context:nil error:nil];
+        }
+        
+        NSString *clientCapabilitiesStr = [json msidStringObjectForKey:MSID_BROKER_CLIENT_CAPABILITIES_KEY];
+        _clientCapabilities = [clientCapabilitiesStr componentsSeparatedByString:@","];
+        
+        NSString *claimsStr= [json msidStringObjectForKey:MSID_BROKER_CLAIMS_KEY];
+        if (claimsStr)
+        {
+            _claimsRequest = (MSIDClaimsRequest *)[[MSIDJsonSerializer new] fromJsonString:claimsStr ofType:MSIDClaimsRequest.class context:nil error:nil];
```

Comment:
nit: can we log an error for debug purposes?

Reply:
We log in json serializer.